### PR TITLE
FIX/PLF-8228: XSS vulnerability in task with variable DueCategory

### PR DESCRIPTION
**Pull request by lbengayed**
This is a proposed fix. It solves the XSS issue PLF-8228.
With this PR, we try to escape  special character "quot" that can caused the XSS vulnerability in task with variable DueCategory.

### CODE REVIEW

File: task-management/src/main/java/org/exoplatform/task/management/controller/TaskController.java [hunk 1]
```diff
@@ -820,7 +820,7 @@ public Response listTasks(String space_group_id, Long projectId, Long labelId, S
         .showCompleted(advanceSearch && filter.isShowCompleted())
         .groupBy(groupBy == null ? TaskUtil.NONE : groupBy)
         .orderBy(orderBy == null ? "" : orderBy)
-        .filter(dueCategory == null ? "" : dueCategory)
+        .filter(dueCategory == null ? "" : dueCategory.replace("\"","&quot;"))
```
Comment (thomasdelhomenie) [inline task-management/src/main/java/org/exoplatform/task/management/controller/TaskController.java]: We should use org.apache.commons.lang3.StringEscapeUtils.escapeHtml() IMO, to make sure the string is correctly escaped and safe.
Also, it should be done only when displaying the value in HTML. For example here for "filter", it is used in the groovy template projectTaskListView.gtmpl, "filter" is used a first time in the method ProjectUtil.permalink, and a second time in the HTML `<input type="hidden" name="filter" value="${filter}"/>`. For the first time it should not be escaped since the method may want to have to have the original value, but it should be escaped in the second case.

File: task-management/src/main/java/org/exoplatform/task/management/templates/projectTaskListView.gtmpl [hunk 2]
```diff
@@ -28,6 +28,7 @@
 	import org.exoplatform.commons.utils.HTMLEntityEncoder;
     import org.exoplatform.task.management.model.ViewType;
     import org.exoplatform.task.util.ProjectUtil;
+    import org.apache.commons.lang.StringEscapeUtils;
```
Comment (thomasdelhomenie) [inline task-management/src/main/java/org/exoplatform/task/management/templates/projectTaskListView.gtmpl]: We should use the version 3 of commons lang, it is newer : org.apache.commons.lang3.StringEscapeUtils

File: task-management/src/main/java/org/exoplatform/task/management/templates/projectTaskListView.gtmpl [hunk 3]
```diff
@@ -69,7 +70,7 @@
                          <input type="text"  name="keyword" value="${keyword}" placeholder="&{label.search}" >
                          <input type="hidden" name="groupBy" value="${groupBy}"/>
                          <input type="hidden" name="orderBy" value="${orderBy}"/>
-                         <input type="hidden" name="filter" value="${filter}"/>
+                         <input type="hidden" name="filter" value="${StringEscapeUtils.escapeHtml(filter)}"/>
```
Comment (thomasdelhomenie) [inline task-management/src/main/java/org/exoplatform/task/management/templates/projectTaskListView.gtmpl]: Do you plan to do another PR for https://jira.exoplatform.org/browse/PLF-8229 ? I think we could fix both issues in the same PR, it is closely related. WDYT ?

File: task-management/src/main/java/org/exoplatform/task/management/controller/TaskController.java (hunk 1 above)
Comment (lbengayed) [reply task-management/src/main/java/org/exoplatform/task/management/controller/TaskController.java]: ok

File: task-management/src/main/java/org/exoplatform/task/management/templates/projectTaskListView.gtmpl (hunk 2 above)
Comment (lbengayed) [reply task-management/src/main/java/org/exoplatform/task/management/templates/projectTaskListView.gtmpl]: ok

File: task-management/src/main/java/org/exoplatform/task/management/templates/projectTaskListView.gtmpl (hunk 3 above)
Comment (lbengayed) [reply task-management/src/main/java/org/exoplatform/task/management/templates/projectTaskListView.gtmpl]: ok
